tests/MyGame/Example/Monster.java

```diff
@@ -344,5 +347,372 @@ public static Monster __lookup_by_key(Monster obj, int vectorLocation, String ke
     public Monster getByKey(String key) {  return __lookup_by_key(null, __vector(), key, bb); }
     public Monster getByKey(Monster obj, String key) {  return __lookup_by_key(obj, __vector(), key, bb); }
   }
-}
-
+  public MonsterT unpack() {
+    MonsterT _o = new MonsterT();
+    this.unpackTo(_o);
+    return _o;
+  }
+  public void unpackTo(MonsterT _o) {
+    MyGame.Example.Vec3T _oPos = this.pos() != null ? this.pos().unpack() : null;
+    _o.setPos(_oPos);
+    short _oMana = this.mana();
+    _o.setMana(_oMana);
+    short _oHp = this.hp();
+    _o.setHp(_oHp);
+    String _oName = this.name();
+    _o.setName(_oName);
+    List<Integer> _oInventory = new ArrayList<>();
```

Comment:
This seems very expensive? In the original, these are a vector of bytes, and here `Integer` is an `Object` so this will be super expensive in comparison.

I would vote to model FlatBuffers vectors in Java as arrays, instead of `List` or `Vector` or whatever, since those really don't work for scalars. That has the disadvantage that if someone wants to resize the array that it is more work, but it is vastly more efficient for the cases where the size stays the same.

We could consider handy helper methods to resize arrays if there's none in the std library.

Reply:
We could also do scalar vectors as arrays, and object vectors as `Vector`, but really it be neater to have them all as arrays.

Reply:
I think that in case of ObjectAPI we can be a bit more permissive with memory consumption, to get more convenient API.
Arrays are fine, but less comfort to work with.
I will start to replace lists with arrays, but take into account that byte to Integer translation price is mainly to save of conversions and missing unsigned variables in java, so in best case you get int[] instead of List<Integer> where the "big" save comes from the array to list.

Reply:
Aside from the already mentioned boxing, with primitive arrays you give opportunity to the VM to remove bound checks, and this really has an impact on performance.

Reply:
Yes, the Object API is more about convenience than the performance oriented base API, but allocating an object for every single byte is excessive. This is still a library about efficiency.

That, and if I want to pass my scalars to other APIs, they're more likely to take arrays. I certainly hope no Java API explicitly takes a `List<Integer>` as an argument..

Reply:
I moved to arrays.
Will share the code in the next PR iteration.